extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -56,35 +57,52 @@ public static function render( $input, array $args, Parser $parser, PPFrame $fra
 		if ( F::app()->checkSkin( 'wikiamobile' ) ) {
 			// In Mercury, discussions are rendered client side as an Ember component
 			$modelData = [
-				'show' => $showLatest ? 'latest' : 'trending',
-				'itemCount' => $itemCount,
+				'mercuryComponentAttrs' => json_encode( [
+					'category' => $category,
+					'show' => $showLatest ? 'latest' : 'trending',
+					'itemCount' => $itemCount,
+				] ),
+				'loading' => wfMessage( 'embeddable-discussions-loading' )->plain()
 			];
 
-			// In mercury, discussions app is rendered client side
+			// In mercury, discussions app is rendered client side in an Ember container
 			$html = $templateEngine->clearData()
 				->setData( $modelData )
 				->render( 'DiscussionThreadMobile.mustache' );
 
 			return $html;
 		} else {
-			$modelData = ( new DiscussionsThreadModel( $wgCityId ) )->getData( $showLatest, $itemCount );
+			$modelData = ( new DiscussionsThreadModel( $wgCityId ) )->getData( $showLatest, $itemCount, $category );
+
+			$modelData['requestData'] = json_encode( [
+				'baseUrl' => $modelData['baseUrl'],
+				'category' => $category,
+				'columns' => $columns,
+				'columnsDetailsClass' => $columns === 2 ? 'embeddable-discussions-post-detail-columns' : '',
+				'showLatest' => $showLatest,
+				'upvoteRequestUrl' => $modelData['upvoteRequestUrl'],
+			] );
+
+			if ( $showLatest && $category ) {
+				$heading = wfMessage( 'embeddable-discussions-show-latest-in-category', $category )->plain();
+			} else if ( $showLatest ) {
+				$heading = wfMessage( 'embeddable-discussions-show-latest' )->plain();
+			} else if ( $category ) {
+				$heading = wfMessage( 'embeddable-discussions-show-trending-in-category', $category )->plain();
+			} else {
+				$heading = wfMessage( 'embeddable-discussions-show-trending' )->plain();
+			}
 
 			$modelData['columns'] = $columns;
 			$modelData['columnsWrapperClass'] = $columns === 2 ? 'embeddable-discussions-threads-columns' : '';
 			$modelData['columnsDetailsClass'] = $columns === 2 ? 'embeddable-discussions-post-detail-columns' : '';
-			$modelData['heading'] = $showLatest ? wfMessage( 'embeddable-discussions-show-latest' )->plain() :
-				wfMessage( 'embeddable-discussions-show-trending' )->plain();
-			$modelData['replyText'] = wfMessage( 'embeddable-discussions-reply' )->plain();
-			$modelData['shareText'] = wfMessage( 'embeddable-discussions-share' )->plain();
+			$modelData['heading'] = $heading;
 			$modelData['showAll'] = wfMessage( 'embeddable-discussions-show-all' )->plain();
-			$modelData['showLatest'] = $showLatest;
-			$modelData['upvoteText'] = wfMessage( 'embeddable-discussions-upvote' )->plain();
-			$modelData['zeroText'] = wfMessage( 'embeddable-discussions-zero' )->plain();
-			$modelData['zeroTextDetail'] = wfMessage( 'embeddable-discussions-zero-detail' )->plain();
+			$modelData['loading'] = wfMessage( 'embeddable-discussions-loading' )->plain();
 
 			$html = $templateEngine->clearData()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)